stock_average_daily_sale/models/stock_average_daily_sale.py

```diff
@@ -308,13 +413,19 @@ def _create_materialized_view(self):
                         GREATEST(
                             (cfg.number_days_qty_in_stock * average_qty_by_sale * average_daily_sales_count) + (ds.daily_standard_deviation * cfg.safety_factor * sqrt(nbr_days)),
                             (cfg.number_days_qty_in_stock *  average_qty_by_sale)
-                        ) as recommended_qty
+                        ) as recommended_qty,
+                        GREATEST(
+                            (cfg.number_days_qty_in_stock * (average_qty_by_sale - COALESCE(average_qty_by_return, 0)) * (average_daily_sales_count - COALESCE(average_daily_returns_count, 0))) + ((ds.daily_standard_deviation - COALESCE(dsr.daily_standard_deviation, 0)) * cfg.safety_factor * sqrt(nbr_days)),
+                            (cfg.number_days_qty_in_stock * (average_qty_by_sale - COALESCE(average_qty_by_return, 0)))
+                        ) as recommended_qty_incl_returns
                     FROM averages t
                     JOIN daily_standard_deviation ds on ds.id= t.window_id
                     JOIN stock_average_daily_sale_config cfg on cfg.id = t.config_id
                     JOIN stock_qty sqty on sqty.pp_id = t.product_id AND t.warehouse_id = sqty.warehouse_id
                     JOIN product_product pp on pp.id = t.product_id
                     JOIN product_template pt on pt.id = pp.product_tmpl_id
+                    LEFT JOIN averages_return tr ON tr.window_id = t.window_id
+                    LEFT JOIN daily_standard_deviation_return dsr on dsr.id= t.window_id
```

Comment:
The `window_id` / `id` is not updated for all of these tables and some of the joins will not match for any record.

These have the `id` still defined as `concat(warehouse_id, product_id)::integer as window_id`
- averages
- daily_standard_deviation

Reply:
Ah, just now noticed the other linked OCA PR that resolves this

Reply:
Indeed, the solution here is only for the returns part, that's why tests are failing here.